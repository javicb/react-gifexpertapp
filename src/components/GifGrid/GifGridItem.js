import React from 'react'

export const GifGridItem = ({ title, url }) => {

   return (
      <>
         <img src={url} alt={title} />
         <p>{title}</p>
      </>
   )
}
