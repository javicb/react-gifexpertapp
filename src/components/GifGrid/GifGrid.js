import React from 'react'
import { useFetchGifs } from '../../hooks/useFetchGifs'
import { GifGridItem } from '../GifGrid/GifGridItem'

export const GifGrid = ({ category }) => {

   const { data: images, loading } = useFetchGifs(category);

   return (
      <>
         <h3 className='animate__animated animate__fadeIn'>{category}</h3>
         {loading && <p className='animate__animated animate__flash'>Cargando...</p>}
         <div className='card-grid'>
            {
               images.map((img) => (
                  <GifGridItem key={img.id} {...img} />
               ))
            }
         </div>
      </>
   )
}
