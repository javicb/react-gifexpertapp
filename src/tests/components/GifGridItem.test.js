import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGrid/GifGridItem';

describe('Pruebas en GifGridItem', () => {
   test('Debe mostrar el componente correctamente ', () => {
      const wrapper = shallow(<GifGridItem />);
      expect(wrapper).toMatchSnapshot();
   })

})
