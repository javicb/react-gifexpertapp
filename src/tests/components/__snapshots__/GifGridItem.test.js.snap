// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridITem Debe mostrar el componente correctamente  1`] = `
<Fragment>
  <div
    className="card animate__animated animate__fadeIn"
  >
    <img />
    <p />
  </div>
</Fragment>
`;
